listenbrainz/webserver/static/js/src/CBReviewModal.tsx

```diff
@@ -0,0 +1,703 @@
+  entityToReview: ReviewableEntity | null;
+  loading: boolean;
+  reviewValidateAlert: string | null;
+
+  releaseGroupEntity: ReviewableEntity | null;
+  artistEntity: ReviewableEntity | null;
+  recordingEntity: ReviewableEntity | null;
+
+  textContent: string;
+  rating: number;
+  language: string;
+  acceptLicense: boolean;
+
+  success: boolean;
+  reviewMBID?: string;
+}
+
+iso.registerLocale(eng); // library requires language of the language list to be initiated
+
+export default class CBReviewModal extends React.Component<
+  CBReviewModalProps,
+  CBReviewModalState
+> {
+  static contextType = GlobalAppContext;
+  declare context: React.ContextType<typeof GlobalAppContext>;
+
+  readonly minTextLength = 25;
+  readonly maxTextLength = 100000;
+
+  private CBBaseUrl = "https://critiquebrainz.org"; // only used for href
+  private MBBaseUrl = "https://metabrainz.org"; // only used for href
+  private allLanguagesKeyValue = Object.entries(iso.getNames("en")); // gets all languages
+
+  private CBInfoButton = (
+    <button
+      type="button"
+      className="btn-transparent capitalize-bold"
+      data-toggle="popover"
+      title="What is CritiqueBrainz?"
+      data-placement="bottom"
+      data-html="true"
+      data-content={`CritiqueBrainz is a <a href='${this.MBBaseUrl}/projects'>
+      MetaBrainz project</a> aimed at providing an open platform for music critics
+      and hosting Creative Commons licensed music reviews. </br></br>
+      Your reviews will be independently visible on CritiqueBrainz and appear publicly 
+      on your CritiqueBrainz profile. To view or delete your reviews, visit your 
+      <a href='${this.CBBaseUrl}'>CritiqueBrainz</a>  profile.`}
+    >
+      {" "}
+      <FontAwesomeIcon icon={faInfoCircle as IconProp} />
+    </button>
+  );
+
+  constructor(props: CBReviewModalProps) {
+    super(props);
+    this.state = {
+      entityToReview: null,
+      loading: false,
+      reviewValidateAlert: null,
+
+      releaseGroupEntity: null,
+      artistEntity: null,
+      recordingEntity: null,
+
+      textContent: "",
+      rating: 0,
+      language: "en",
+      acceptLicense: false,
+
+      success: false,
+    };
+  }
+
+  async componentDidMount() {
+    await this.getAllEntities();
+  }
+
+  async componentDidUpdate(prevProps: CBReviewModalProps) {
+    const { listen } = this.props;
+    if (prevProps.listen !== listen) {
+      this.setState({
+        textContent: "",
+        rating: 0,
+        reviewValidateAlert: null,
+        success: false,
+      });
+
+      await this.getAllEntities();
+    }
+  }
+
+  handleError = (error: string | Error, title?: string): void => {
+    const { newAlert } = this.props;
+    if (!error) {
+      return;
+    }
+    newAlert(
+      "danger",
+      title || "Error",
+      typeof error === "object" ? error.message : error
+    );
+  };
+
+  hasPermissions = (user?: CritiqueBrainzUser) => {
+    if (!user || !user.access_token) {
```

Comment:
This can be simplified.

Reply:
Do you mean this?
```suggestion
    if (!user?.access_token) {
```

Reply:
Yes @MonkeyDo ! :)
Infact, we can make this a one liner as `return !user?.access_token;`

Reply:
I guess we want to invert that boolean: `return !!user?.access_token;`

Reply:
Yasss @MonkeyDo you're on point. A correction regarding that! Thanks.